### ISSUE TITLE
Possible bug

### ISSUE THREAD

**Issue by dajmcdon** [cmu-delphi/epiprocess#314]
https://github.com/cmu-delphi/epiprocess/blame/6b8fad88a09bd79a31fea56a595c7d7df4ea46ff/R/slide.R#L176

This line broke `{epipredict}` vignettes. Suppose one calls `epix_as_of(archive, "2021-01-01")` on an archive that has 1 day latency. The resulting `epi_df` would then have `max(time_value) = "2020-12-31"`. If you pass that to `epi_slide()` and try to make a forecast with forecast data `2021-01-01`, it will Abort.

**Comment by dajmcdon** [comment]
This is on the `dev` branch. On `main`, `epi_slide()` subsets to available time values and ignores that last forecast date. This isn't ideal either, but didn't crash.

**Comment by brookslogan** [comment]
This is when the user is explicitly requesting a value for that `ref_time_value` by passing their own `ref_time_values` rather than relying on the default, right?  I would assume they'd want an error if they're not going to get a value for it?  I think this was the reasoning when we made this into an error.

**Comment by brookslogan** [comment]
Though it might make sense sometimes, especially if we add computation-input completion helpers, to allow ref time values beyond the max time value.  Were you thinking of something like this, where the user will get an NA for values corresponding to time_value 2021-01-01?

**Comment by dajmcdon** [comment]
Yes, explicit `ref_time_values`. In my current example, think NA is the right thing, yes. 

For this exercise, this beahviour seems incongruous with `epix_as_of()`, but maybe it's as expected for something like a 7dsum. (actually, I'm not even sure that's true). From the standpoint of a rolling pseudo-prospective forecaster, I'm imagining that `epix_slide` is handing me the result of the `_as_of()` call for each `ref_time_value` I've requested. But that doesn't happen if I request something at the end of the period. Instead, it just errors out. Though, given 1 day latency, for earlier ref_time_values, I AM getting (essentially) an NA for the ref_time_value.

I almost think that the check should be `ref_time_values >= min(time_value)`.
